<launch>

    <arg name="multi_robot_name" default="" />
    <arg name="set_lidar_frame_id" default="base_scan" />
    <arg name="set_frame_id" default="base_scan"/>
    
    <env name="ROSLAUNCH_SSH_UNKNOWN" value="1" />
    <machine name="robot222" address="robot222.lan" user="robot" env-loader="/home/robot/remote_env.sh"/>

    <node machine="robot222" pkg="rosserial_python" type="serial_node.py" name="turtlebot3_core" output="screen">
        <param name="port" value="/dev/ttyACM0" />
        <param name="baud" value="115200" />
        <param name="tf_prefix" value="$(arg multi_robot_name)" />
    </node>

    <node machine="robot222" pkg="hls_lfcd_lds_driver" type="hlds_laser_publisher" name="turtlebot3_lds" output="screen">
        <param name="port" value="/dev/ttyUSB0" />
        <param name="frame_id" value="$(arg set_frame_id)" />
    </node>

    <node machine="robot222" pkg="raspicam_node" type="raspicam_node" name="raspicam_node" output="screen">
        <param name="camera_info_url" value="package://turtlebot3_bringup/camera_info/turtlebot3_rpicamera.yaml" />
        <param name="width" value="640" />
        <param name="height" value="480" />
        <param name="framerate" value="15" />
        <param name="camera_frame_id" value="camera" />
    </node>

    <node machine="robot222" pkg="turtlebot3_bringup" type="turtlebot3_diagnostics" name="turtlebot3_diagnostics" output="screen" />

</launch>